File: src/ghost_mcp/types/mcp.py
"""MCP-specific type definitions."""

from typing import Any, Dict, List, Optional, Union

from pydantic import BaseModel


class MCPToolParameter(BaseModel):
    """MCP tool parameter definition."""
    type: str
    description: Optional[str] = None
    required: bool = False
    enum: Optional[List[str]] = None
    properties: Optional[Dict[str, "MCPToolParameter"]] = None


class MCPToolDefinition(BaseModel):
    """MCP tool definition."""
    name: str
    description: str
    inputSchema: Dict[str, Any]


class MCPContent(BaseModel):
    """MCP response content."""
    type: str
    text: Optional[str] = None
    data: Optional[str] = None
    url: Optional[str] = None
    mimeType: Optional[str] = None


class MCPToolResponse(BaseModel):
    """MCP tool response."""
    content: List[MCPContent]
    isError: bool = False


class MCPToolRequest(BaseModel):
    """MCP tool request."""
    name: str
    arguments: Dict[str, Any]


class MCPError(BaseModel):
    """MCP error response."""
    code: int
    message: str
    data: Optional[Any] = None

File: src/types/mcp.ts
/**
 * MCP-Specific Types
 *
 * TypeScript interfaces for MCP tool definitions and responses.
 */

// TODO: Phase 1 - Define MCP-specific types

export interface MCPToolParameter {
  type: string;
  description?: string;
  required?: boolean;
  enum?: string[];
  properties?: Record<string, MCPToolParameter>;
}

export interface MCPToolDefinition {
  name: string;
  description: string;
  inputSchema: {
    type: 'object';
    properties: Record<string, MCPToolParameter>;
    required?: string[];
  };
}

export interface MCPToolResponse {
  content: Array<{
    type: 'text' | 'image' | 'resource';
    text?: string;
    data?: string;
    url?: string;
    mimeType?: string;
  }>;
  isError?: boolean;
}

export interface MCPToolRequest {
  name: string;
  arguments: Record<string, unknown>;
}

// MCP Error types
export interface MCPError {
  code: number;
  message: string;
  data?: unknown;
}

// Configuration types
export interface MCPServerConfig {
  ghost: {
    url: string;
    contentApiKey?: string;
    adminApiKey?: string;
    version: string;
    mode: 'readonly' | 'readwrite' | 'auto';
  };
  logging: {
    level: 'debug' | 'info' | 'warn' | 'error';
    structured: boolean;
  };
}